### ISSUE TITLE
Duplicated triples generated in train_test_split_no_unseen()

### ISSUE THREAD

**Issue by pyvandenbussche** [Accenture/AmpliGraph#68]
### Description
The script to carve a train/test set from a list of triples leads to:
- duplicated test triples 
- and therefore the train set being the diff of all triples - test ones is bigger than expected.

Will do a pull request with fix.

**Comment by chanlevan** [comment]
Reopen to make sure the results must be deterministic and there should be a unit test to cover.
